src/fstab-generator/fstab-generator.c

```diff
@@ -1164,4 +1230,18 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
         return r < 0 ? r : r2 < 0 ? r2 : r3;
 }
 
-DEFINE_MAIN_GENERATOR_FUNCTION(run);
+static int run(int argc, char **argv) {
+        arg_sysroot_only = streq_ptr(argv[1], "--sysroot-only");
```

Comment:
shouldn't this check that if true it's running in initrd and return an error if not? There's an assert in `parse_fstab` but that's not very user friendly